system/nagios-status-check/detectors-nagios-status-check.tf

```
@@ -0,0 +1,37 @@
+resource "signalfx_detector" "status_check" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Nagios check status"
+
+  program_text = <<-EOF
+        signal = data('nagios_state.state', filter=${module.filter-tags.filter_custom})${var.status_check_aggregation_function}${var.status_check_transformation_function}.publish('signal')
+        detect(when(signal >= threshold(1)) and when(signal < 2)).publish('WARN')
```

```suggestion
        detect(when(signal == threshold(1))).publish('WARN')
```

should be enough and easier and will allow lasting function